add a script for setting up separate locales

proletarius101 opened the pull request:
See also https://github.com/flatpak/flatpak-builder-tools/pull/341#discussion_r1066494701

proletarius101 committed:
add a script for setting up separate locales

gasinvein commented on org.electronjs.Electron2.BaseApp.yml:
```diff
@@ -85,3 +84,25 @@ modules:
+        dest-filename: setup-separate-locales
+        commands:
+          - |
+            APP_LOCALES_DIR=$1
```
I think it would be better if the script would accept the app root dir as argument, not the locales dirs. The `/locales` subdir is always the same, so it doesn't make much sense to add to every script invocation.
```suggestion
            APP_LOCALES_DIR=$1/locales
```

proletarius101 committed:
accept the app root dir as argument rather than the locales dirs

The `/locales` subdir is always the same, so it doesn't make much sense to add to every script invocation.

Co-authored-by: Filippe LeMarchand <gasinvein@gmail.com>

gasinvein commented on org.electronjs.Electron2.BaseApp.yml:
```diff
@@ -85,3 +84,25 @@ modules:
+        dest-filename: setup-separate-locales
+        commands:
+          - |
+            APP_LOCALES_DIR=$1/locales
+            RUNTIME_LOCALES_DIR=${FLATPAK_DEST}/share/runtime/locale
+            if [ -d ${APP_LOCALES_DIR} ]
+            then
+                for lang in ${APP_LOCALES_DIR}/*.pak
+                do
+                locale="$(basename -s .pak $lang)"
+                install -Dm644 -t "${RUNTIME_LOCALES_DIR}/${locale%%-*}/" "$lang"
+                ln -sf "${RUNTIME_LOCALES_DIR}/${locale%%-*}/$(basename $lang)" "${APP_LOCALES_DIR}/$(basename $lang)"
+                done
+            fi
```
Since electron apps dirs are often named after the app, it's not unlikely for the dir to have spaces in its name, so we should quote all the paths.
```suggestion
            APP_LOCALES_DIR="$1/locales"
            RUNTIME_LOCALES_DIR="${FLATPAK_DEST}/share/runtime/locale"
            if [ -d "${APP_LOCALES_DIR}" ]
            then
                for lang in "${APP_LOCALES_DIR}"/*.pak
                do
                locale="$(basename -s .pak "$lang")"
                install -Dm644 -t "${RUNTIME_LOCALES_DIR}/${locale%%-*}/" "$lang"
                ln -sf "${RUNTIME_LOCALES_DIR}/${locale%%-*}/$(basename "$lang")" "${APP_LOCALES_DIR}/$(basename "$lang")"
                done
            fi
```

refi64 commented on org.electronjs.Electron2.BaseApp.yml:
```diff
@@ -85,3 +84,25 @@ modules:
+      - type: script
+        dest-filename: setup-separate-locales
+        commands:
+          - |
```
nit: `set -e` at the top

refi64 commented on org.electronjs.Electron2.BaseApp.yml:
```diff
@@ -85,3 +84,25 @@ modules:
+            then
+                for lang in ${APP_LOCALES_DIR}/*.pak
+                do
+                locale="$(basename -s .pak $lang)"
```
Should use a new indentation level here.

refi64 commented on org.electronjs.Electron2.BaseApp.yml:
```diff
@@ -85,3 +84,25 @@ modules:
+                for lang in ${APP_LOCALES_DIR}/*.pak
+                do
+                locale="$(basename -s .pak $lang)"
+                install -Dm644 -t "${RUNTIME_LOCALES_DIR}/${locale%%-*}/" "$lang"
```
nit: could save the first path into an intermediate variable to avoid repeating it

proletarius101 committed:
improve robustness and code style

proletarius101 commented:
Hi @refi64. Any updates?

refi64 commented:
oh I was hoping @gasinvein would give it another round? (and then...forgot and merged https://github.com/flatpak/flatpak-builder-tools/pull/341 anyway, oops...)

bbhtt commented:
Is there anything more to do here?
